tutorials/high_performance_networking/README.md

```diff
@@ -0,0 +1,2179 @@
+10. The size of each buffer in the memory region. These should be equal to your maximum packet size, or less if breaking down packets (ex: header data split, see the `rx` queue below).
+11. The `interfaces` section lists the NIC interfaces that will be configured for the application.
+12. A descriptive name for that interface, currently only used for logging.
+13. The PCIe/bus address of that interface, as identified in previous sections.
+14. Each interface can have a `tx` (transmitting) or `rx` (receiving) section, or both if you'd like to configure both Tx and Rx on the same interface.
+15. The `queues` section lists the queues for that interface. Queues are a core concept of NICs: they handle the actual receiving or transmitting of network packets. Rx queues buffer incoming packets until they can be processed by the application, while Tx queues hold outgoing packets waiting to be sent on the network. The simplest setup uses only one receive and one transmit queue. Using more queues allows multiple streams of network traffic to be processed in parallel, as each queue can be assigned to a specific CPU core, and are assigned their own memory regions that are not shared.
+16. A descriptive name for that queue, currently only used for logging.
+17. The ID of that queue, which can be referred to later in the `flows` section.
+18. The number of packets per batch. The `advanced_network` Rx and Tx operators will "tick" for each batch.
+19. The ID of the CPU core that this queue will use to poll the NIC. Ideally one [isolated core](#35-isolate-cpu-cores) per queue.
+20. The list of memory regions where this queue will write/read packets from/to. The order matters: the first memory region will be used first to read/write from until it fills up one buffer (`buf_size`), after which it will move to the next region in the list and so on until the packet is fully written/read. See the `memory_regions` for the `rx` queue below for an example.
+21. Same as for `data1`. Each interface in this list should have a unique mac address. This one will do `rx` per config below.
+22. Whether to isolate the Rx flow. If true, any incoming packets that does not match the MAC address of this interface - or isn't directed to a queue when the `flows` section below is used - will be delegated back to Linux for processing (no kernel bypass). This is useful to let this interface handle ARP, ICMP, etc. Otherwise, any packets sent to this interface (ex: ping) will need to be processed (or dropped) by your application.
+23. `rx` queues have an `output_port` parameter so you can attach a downstream operator to receive data from this specific queue, as can be seen in the `#!cpp Application::compose()` function of the sample application. Multiple `rx` queues can share the same `output_port`. In contrast, `tx` queues have a single non-configurable port (name: `burst_in`) to which upstream operators will send all packets, which are then routed to the correct queue based on #TODO#.
```

Comment:
based on the port/queue in the burst header

Reply:
the part I still can't grasp, is how is the port/queue defined in the burst header?

I assume port is inferred from the app passing the source mac address (defined below in our config file). Is that correct? and how is queue inferred?